pkg/controller/populators/clone-populator.go

```diff
@@ -302,8 +302,13 @@ func (r *ClonePopulatorReconciler) validateCrossNamespace(pvc *corev1.Persistent
 
 func (r *ClonePopulatorReconciler) reconcileDone(ctx context.Context, log logr.Logger, pvc *corev1.PersistentVolumeClaim) (reconcile.Result, error) {
 	log.V(3).Info("executing cleanup")
-	if err := r.planner.Cleanup(ctx, log, pvc); err != nil {
-		return reconcile.Result{}, err
+	if pvc.Annotations[cc.AnnPodRetainAfterCompletion] == "true" {
+		// Avoiding cleanup so we can keep clone objects for debugging purposes.
+		r.recorder.Eventf(pvc, corev1.EventTypeWarning, retainedPVCPrime, messageRetainedPVCPrime)
+	} else {
+		if err := r.planner.Cleanup(ctx, log, pvc); err != nil {
```

Comment:
Hey @mhenriks, what do you think of skipping the cleanup when using `AnnPodRetainAfterCompletion`? I think that for debugging purposes we can allow keeping all the clone objects (even the PVC prime in lost state).

Another possibility is to have the PVC Prime in terminating state until the user deletes the pod, but having the PVC stuck there seems awkward to me. Maybe we could also consider removing the PVC protection finalizer to delete the PVC prime.

Reply:
Only do this for host clone

Reply:
One thing to note here though.  Pods will be in the source namespace.  With cross namespace clone, the user may not have access to them.  And I'm not sure we want to allow the user to do this in cross namespace case

Reply:
Yeah that's a good point. What about the check in the DV controller to see if we use populators or not? 

https://github.com/kubevirt/containerized-data-importer/blob/bd6335259dd12ba518634b2366be5c664567ba40/pkg/controller/datavolume/controller-base.go#L1180-L1184

Are we ok with turning to populators even if we only use `AnnPodRetainAfterCompletion` with host assisted clone?

Reply:
I guess we are since it's basically the same behavior without populators.

Reply:
Without populators, the target pod is in the target namespace.  With populators both pods are in the source then rebind happens.  The reason for having the target pod in the source is to better support k8s native cross namespace auth (when it goes live).  In that case there will be no clone token.  So can't do cross namespace host clone